ui/common/Input.tsx

```diff
@@ -1,39 +1,93 @@
-'use client'
-
-import RICIBs from 'react-individual-character-input-boxes'
-
-export default function Input({
-  amount,
-  onChange,
-}: {
-  amount: number
-  onChange: Function
-}) {
-  const handleChange = (value: string) => {
-    onChange(value)
-  }
-
-  return (
-    <RICIBs
-      amount={amount}
-      autoFocus
-      handleOutputString={handleChange}
-      inputProps={{
-        className: 'bg-transparent',
-        placeholder: '_',
-        style: {
-          fontSize: '20px',
-          width: '20px',
-          height: '20px',
-          margin: '0px',
-          borderRadius: '0px',
-          textAlign: 'center',
-          justifyContent: 'space-evenly',
-          outline: 'none',
-          fontFamily: 'var(--space-mono-font)',
-        },
-      }}
-      inputRegExp={/^[a-zA-Z0-9_.-]*$/}
-    />
-  )
-}
+'use client'
+
+import { useState } from 'react'
+
+interface UserInputProps {
+  onChange: Function
+  answer: string
+  hints?: boolean
+}
+
+export default function Input({ onChange, answer, hints }: UserInputProps) {
+  const [textAreaValue, setTextAreaValue] = useState('')
+  const [correctAns, setCorrectAns] = useState(false)

```

Comment:
Nitpick but let's try to steer away from abbreviations like ans for answer, these might be confusing for non-english folk and could leave some room for interpretation, which we want to avoid.